From: Or Gerlitz <ogerlitz@mellanox.com>
Date: Fri, 10 Jun 2016 00:07:34 +0300
Subject: [PATCH] net/mlx5: E-Switch, Use the correct free() function
Patch-mainline: v4.7-rc3
Git-commit: 3fe3d819d5015d56d0d7289ae16db5e612640c5b (partial)
References: bsc#966170 FATE#320225 bsc#966172 FATE#320226

We must use kvfree() for something that could have been allocated with vzalloc(),
do that.

Fixes: 5742df0f7dbe ('net/mlx5: E-Switch, Introduce VST vport ingress/egress ACLs')
Fixes: 86d722ad2c3b ('net/mlx5: Use flow steering infrastructure for mlx5_en')
Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
Reported-by: Ilya Lesokhin <ilyal@mellanox.com>
Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/mellanox/mlx5/core/eswitch.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/mellanox/mlx5/core/eswitch.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/eswitch.c
@@ -426,7 +426,7 @@ static int esw_create_fdb_table(struct m
 	esw->fdb_table.addr_grp = g;
 	esw->fdb_table.fdb = fdb;
 out:
-	kfree(flow_group_in);
+	kvfree(flow_group_in);
 	if (err && !IS_ERR_OR_NULL(fdb))
 		mlx5_destroy_flow_table(fdb);
 	return err;
